File: clients/csharp/Src/elencyConfig/LocalConfiguration.cs
using System;
using System.Collections.Generic;

namespace ElencyConfig
{
    public class LocalConfiguration
    {
        public string AppVersion { get; set; }
        public string Environment { get; set; }
        public string ConfigurationId { get; set; }
        public Dictionary<string, string> ConfigurationData { get; set; }

        internal void Validate()
        {
            if (string.IsNullOrWhiteSpace(AppVersion))
            {
                throw new Exception("AppVersion has not been defined on LocalConfiguration");
            }

            if (string.IsNullOrWhiteSpace(Environment))
            {
                throw new Exception("Environment has not been defined on LocalConfiguration");
            }

            if (string.IsNullOrWhiteSpace(ConfigurationId))
            {
                throw new Exception("ConfigurationId has not been defined on LocalConfiguration");
            }

            if (ConfigurationData == null)
            {
                throw new Exception("configurationData has not been defined on LocalConfiguration");
            }
        }
    }
}

File: clients/ruby/lib/ElencyConfig/LocalConfiguration.rb
require 'net/http'

class Configuration

  def initialize()
    @appVersion = nil
    @environment = nil
    @configurationId = nil
    @configurationData = nil
  end

  def appVersion
    @appVersion
  end

  def appVersion=(appVersion)
    @appVersion
  end

  def environment
    @environment
  end

  def environment=(environment)
    @environment
  end

  def configurationId
    @configurationId
  end

  def configurationId=(configurationId)
    @configurationId
  end

  def configurationData
    @configurationData
  end

  def configurationData=(configurationData)
    @configurationData
  end

  def validate
    if (@appVersion == nil)
      raise "appVersion has not been defined on LocalConfiguration"
    end

    if (@environment == nil)
      raise "environment has not been defined on LocalConfiguration"
    end

    if (@configurationId == nil)
      raise "configurationId has not been defined on LocalConfiguration"
    end

    if (@configurationData == nil)
      raise "configurationData has not been defined on LocalConfiguration"
    end

  end

end